app/generate-badges.py

```diff
@@ -109,8 +127,18 @@ def generate_badges(aggregate, folder, index, picture, paper_size):
             if options['badge_wrap']:
                 aggregate.append(row)
             if len(aggregate) >= NUMBER_OF_BADGES_PER_PAGE:
-                generate_badges(aggregate, folder, i, badges_background, options)
+                if font_choice != '':
+                    generate_badges(aggregate, folder, i,
+                                    badges_background, options, font=font_choice)
+                else:
```

Comment:
@yashLadha you can avoid the else part here by simply checking in the generate_badges method whether fonts variable is an empty string or not